### PR TITLE
add utilities for feedback types and subtypes

### LINKED ISSUE #4799: Feedback: separation feedback types(categories) and desctriptions(sub-categories)
[FeedbackEvent](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackEvent.kt) provides a very poor user experience as it's not clear which Feedback type should have sub-types and which feedback might be used in ActiveGuidance/FreeDrive/Arrival.
We have to provide methods to get:
- divide feedback types on FreeDrive/AcitveGuidance/Arrival;
- provide a set of sub-types for each feedback type.

cc @Zayankovsky @abhishek1508

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description
Closes #4799. 

### Changelog

```
<changelog>Renamed `FeedbackEvent.Description` annotation to `FeedbackEvent.SubType`</changelog>
<changelog>Added `FeedbackHelper` with utilities for working with feedback types and subtypes</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackEvent.kt [hunk 1]
```diff
@@ -249,6 +249,14 @@ object FeedbackEvent {
     /**
      * Type of feedback mean WHAT happen
      */
+    @Target(
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackEvent.kt:252]: 👍

**Comment by LukasPaczos** [comment]
Could you mention in the changelog entry where the new utilities can be found and what is the scope of breaking changes?

**Comment by Zayankovsky** [comment]
> Could you mention in the changelog entry where the new utilities can be found and what is the scope of breaking changes?

Done.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 2]
```diff
@@ -912,15 +912,15 @@ class MapboxNavigation(
      * @param description description message
      * @param feedbackSource one of [FeedbackEvent.Source]
      * @param screenshot encoded screenshot (optional)
-     * @param feedbackSubType array of [FeedbackEvent.Description] (optional)
+     * @param feedbackSubType array of [FeedbackEvent.SubType] (optional)
      */
     @JvmOverloads
     fun postUserFeedback(
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:918]: could you also mention here how to get relevant `subType`(s) fo aimed `type`

**Comment by RingerJK** [comment]
for visibility @azarovalex

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+ * Provides utilities for working with feedback types and subtypes
+ */
+object FeedbackHelper {
+
+    /**
+     * Returns feedback types to send during Free Drive.
+     *
+     * @return array of free drive feedback types
+     */
+    fun getFreeDriveFeedbackTypes(): Array<@FeedbackEvent.Type String> {
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt:13]:
```suggestion
    @JvmStatic
    fun getFreeDriveFeedbackTypes(): Array<@FeedbackEvent.Type String> {
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            FeedbackEvent.OTHER_ISSUE,
+        )
+    }
+
+    /**
+     * Returns feedback types to send during Active Navigation.
+     *
+     * @return array of active navigation feedback types
+     */
+    fun getActiveNavigationFeedbackTypes(): Array<@FeedbackEvent.Type String> {
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt:28]:
```suggestion
    @JvmStatic
    fun getActiveNavigationFeedbackTypes(): Array<@FeedbackEvent.Type String> {
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            FeedbackEvent.POSITIONING_ISSUE,
+        )
+    }
+
+    /**
+     * Returns feedback types to send upon arrival at the destination point.
+     *
+     * @return array of arrival feedback types
+     */
+    fun getArrivalFeedbackTypes(): Array<@FeedbackEvent.Type String> {
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt:44]:
```suggestion
    @JvmStatic
    fun getArrivalFeedbackTypes(): Array<@FeedbackEvent.Type String> {
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /**
+     * Returns feedback subtypes that can be sent with a given [feedbackType].
+     *
+     * @param feedbackType feedback type to send
+     * @return array of feedback subtypes for a given [feedbackType], may be empty
+     * @throws IllegalArgumentException if [feedbackType] is unknown
+     */
+    fun getFeedbackSubTypes(
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackHelper.kt:58]:
```suggestion
    @JvmStatic
    fun getFeedbackSubTypes(
```

**Review by RingerJK — APPROVED** [review]
add please `@JvmStatic` to support java users and other LGTM 👍

**Pull request merged**